### PR TITLE
Report correct exit code on default signal handling

### PR DESCRIPTION
**Pull request by jxyang**
Signed-off-by: Xuejun Yang <xuejya@microsoft.com>

### CODE REVIEW

**Review by salsal97 — APPROVED** [review]
lgtm

File: kernel/signal.c [hunk 1]
```diff
@@ -155,7 +155,7 @@ static long _default_signal_handler(unsigned signum)
             myst_tcall_wake(process_thread->event);
         }
     }
-    thread->exit_status = 0;
+    thread->exit_status = 128 + signum;
```
Comment (mikbras) [inline kernel/signal.c:158]: What is the significance of 128?

**Review by mikbras — APPROVED** [review]

**Comment by mikbras** [comment]
I tried this and it resolves https://github.com/deislabs/mystikos/issues/567.

File: kernel/signal.c (hunk 1 above)
Comment (mikbras) [reply kernel/signal.c:158]: Can you use _NSIG here?

Comment (jxyang) [reply kernel/signal.c:158]: The magic number is from here: https://www.gnu.org/software/bash/manual/html_node/Exit-Status.html. There seems to be no macro for it.

**Pull request merged**